recipe-server/normandy/recipes/tests/__init__.py

```diff
@@ -108,6 +106,20 @@ def locales(self, create, extracted, **kwargs):
             for locale in extracted:
                 self.locales.add(locale)
 
+    # This should always be before `enabled`
+    @factory.post_generation
+    def approver(self, create, extracted, **kwargs):
+        if extracted:
+            approval = ApprovalRequestFactory(revision=self.latest_revision)
+            approval.approve(extracted, 'r+')
+
+    # This should always be after `approver` as we require approval to enable a recipe
```

Comment:
Can we guarantee that these two methods are executed in order? That seems a really finnicky thing to depend on.

I'm still not convinced that parameters to the factory are the best way to go about this. Do you have a reason to use them over something else?